### ISSUE TITLE
Node not get connected to all known nodes

### ISSUE THREAD

**Issue by andyceo** [ergoplatform/ergo#145]
`/peers/all` response show:

```
[
  {
    "address": "159.203.94.149/159.203.94.149:9001",
    "lastSeen": 1518090270055,
    "connectionType": "outgoing"
  },
  {
    "address": "139.59.254.126/139.59.254.126:9001",
    "lastSeen": 1518090270055,
    "connectionType": "outgoing"
  },
  {
    "address": "static.88-198-13-202.clients.your-server.de/88.198.13.202:9001",
    "lastSeen": 1518090270055,
    "connectionType": "outgoing"
  }
]
```

but `peers/connected` show only this:

```
[
  {
    "address": "None",
    "name": "DO-139.59.254.126",
    "lastSeen": 1518090086364
  },
  {
    "address": "None",
    "name": "ergo-seed",
    "lastSeen": 1518090086364
  }
]
```

anwer from `info` shows that `"peersCount" : 2,`

Seems like something going wrong behaviour...

**Comment by daron666** [comment]
Just tried to reproduce.
Resp from `peers/all`
```
[
  {
    "address" : "159.203.94.149/159.203.94.149:9001",
    "lastSeen" : 1521737955626,
    "connectionType" : "Outgoing"
  },
  {
    "address" : "139.59.254.126/139.59.254.126:9001",
    "lastSeen" : 1521737955626,
    "connectionType" : "Outgoing"
  },
  {
    "address" : "static.239.93.46.78.clients.your-server.de/78.46.93.239:9001",
    "lastSeen" : 1521737936155,
    "connectionType" : "Outgoing"
  },
  {
    "address" : "static.88-198-13-202.clients.your-server.de/88.198.13.202:9001",
    "lastSeen" : 1521737955626,
    "connectionType" : "Outgoing"
  }
]
```
Resp from `peers/connected`
```
[
  {
    "address" : "",
    "lastSeen" : 1521737975877,
    "name" : "DO-159.203.94.149"
  },
  {
    "address" : "",
    "lastSeen" : 1521737975877,
    "name" : "DO-139.59.254.126"
  },
  {
    "address" : "",
    "lastSeen" : 1521737975877,
    "name" : "catena-personal"
  },
  {
    "address" : "",
    "lastSeen" : 1521737975877,
    "name" : "ergo-seed"
  }
]
```
Resp from `info`
```
{
  "currentTime" : 1521737972375,
  "name" : "ergo-testnet",
  "stateType" : "utxo",
  "difficulty" : null,
  "votes" : "UAj1rAf",
  "bestFullHeaderId" : null,
  "bestHeaderId" : "5c4QzGRgcv64mwWJVgAJRtWfN16dZ2RMA2VxQcAWisV",
  "peersCount" : 4,
  "unconfirmedCount" : 0,
  "appVersion" : "0.2.3",
  "stateRoot" : null,
  "previousFullHeaderId" : null,
  "fullHeight" : null,
  "headersHeight" : 199,
  "stateVersion" : null,
  "launchTime" : 1521737923470,
  "isMining" : false
}
```

Looks like it's working. @andyceo could you please confirm that issue still exists?

**Comment by andyceo** [comment]
Hi!

Here is a hypothetical steps to reproduce:

1. Let suppose we have enabled and operational nodes which ips and ports are in ergo node sources:

        node_ip_in_sources1
        node_ip_in_sources2

2. Add node_behind_nat1 to the network:

        node_ip_in_sources1
        node_ip_in_sources2
        |
        |-> node_behind_nat1 (once enabled, see only node_ip_in_sources{1,2}) 

3. Next, we enable another opened node, so net topology become to:

        node_ip_in_sources1
        node_ip_in_sources2
        |
        |-> node_behind_nat1 (once enabled, see only node_ip_in_sources{1,2}, after node1 is enabled, should see and connect to node1 but it does not)
        |
        |-> node1 (once enabled, see only node_ip_in_sources{1,2} and should not see node_behind_nat1)

4. And finally we add one more node behind nat, so net topology now is:

        node_ip_in_sources1
        node_ip_in_sources2
        |
        |-> node_behind_nat1 (once enabled, see only node_ip_in_sources{1,2})
        |
        |-> node1 (once enabled, see only node_ip_in_sources{1,2}, after node discovery, should not see node_behind_nat1)
        |
        |-> node_behind_nat2 (once enabled, see only node_and_in_sources{1,2}, after node discovery, should see node1 and connect to it, but it does not)

Hope this help to resolve this issue.

PS: I am not sure which case exactly (3 or 4 or both) I observed.

**Comment by catena2w** [comment]
should be already fixed in v1.7
